app/main/controllers/template/RTMediaTemplate.php

```diff
@@ -219,6 +219,12 @@ function json_output() {
 		$media_array = array();
 
 		if ( $rtmedia_query->media ) {
+
+			// Remove all filters used for search functionality to avoid modifying query for current media results.
+			remove_filter( 'rtmedia-model-where-query', 'rtmedia_search_fillter_where_query', 10, 3 );
+			remove_filter( 'rtmedia-model-join-query', 'rtmedia_search_fillter_join_query', 11, 2 );
+			remove_filter( 'rtmedia-model-query-columns', 'rtmedia_model_query_columns', 10, 1 );
+
```

Comment:
@kiranpotphode I think we should do add_filter for all three filters after foreach loop is completed.